tests/jballerina-unit-test/src/test/resources/test-src/functions/functions_with_default_parameters.bal

```diff
@@ -300,6 +300,24 @@ function value2(record {|int a; int b;|} x) returns int {
     return x.a + x.b;
 }
 
+type F1 function() returns int;
+
+type F2 function() returns function() returns int;
+
+function baz(int a, F1 b = function() returns int { return a + 1; }) returns int {
+    return a + b();
+}
+
+function baz1(int a, F2 b = function() returns function() returns int { return function () returns int { return a; }; }) returns int {
```

Comment:
Can we have more nested tests with scenarios like

```ballerina
function foo(int a, function() returns int b = function() returns int { 
        function (int x = a) returns int f = function (int p) returns  int {
            return p + 1;
        };
        return f(); 
    }
) {
    ...
}
```

Reply:
Add unit tests